### ISSUE TITLE
Unverified channels and no channel with name or ID exists in guild?

### ISSUE THREAD

**Issue by mixels** [Eco-DiscordLink/EcoDiscordPlugin#69]
I'm struggling to understand a problem I'm having with this mod. When the server initializes, I see INFO messages in the console indicating "unverified channels", a concept I can't understand... What do these console messages mean, or would it be possible to update project documentation to explain this terminology?

```
[07:02:10.782] [29] [ Info] [Eco] [DiscordLink] Unverified channels detected:
691069113669255229 - 963882919485636728 <--> General
691069113669255229 - https://discord.com/channels/691069113669255229/963883347979952218
691069113669255229 - 963883259983454218
691069113669255229 - 963883285908451328
691069113669255229 - 963885704734212146
691069113669255229 - 963883385346990151
691069113669255229 - 963883441458384946
691069113669255229 - 963883492431769680 
```

Above you see the error occurring with channel IDs. I see the same messages and also the below error when using channel names rather than IDs. (I tried both ways.)

Also, what is perhaps an actual functional problem or perhaps a misconfiguration, I am seeing the below message whenever a player chats in-game. To address it, I have double checked permissions for the bot on Discord and confirmed the bot member in Discord definitely does have View Channel and Send Message permissions to both the channel and the channel. I'm struggling to think of other ideas as to what could cause this issue. I'd greatly appreciate some help!

`[09:15:40.481] [329] [ Info] [Eco] [DiscordLink] Failed to forward Eco message from user Trenyc as no channel with the name or ID 963882919485636728 exists in the guild `

**Comment by DarkMukke** [comment]
PR #70 fixes this but you'll have to compile the DDLs yourself, I could create a zipfile of the modfile DDLs if you want and put it up as a release on my fork, let me know if you'll be able to compile them yourself

**Comment by mixels** [comment]
Than you, I can build it myself. Will give it's try and let you know.

**Comment by mixels** [comment]
Ok built and tried it out. The unverified channels messages are gone now, but I noticed a few problematic things.

First, the config contract seems to have changed. Before, the Discord server ID was pulled from each channel's property "DiscordGuild". In this branch, I get a message "Unable to find Discord server with name ''", which seems to stem from the code looking for a top level property "DiscordServer". This property didn't exist in my old config.

Second, after adding the top-level "DiscordServer" property, I no longer see any messages at all from DiscordLink in the console output on server start. That said, I also do not see any messages from Eco in my configured channels. I'm not sure what else to say since there's nothing in server logs from the mod... It just doesn't work.

Hope this helps.

**Comment by MonzUn** [comment]
@mixels The PR @DarkMukke references is an update to 2.2.3 for 9.4.6 compatibility and compatibility with API changes made by Discord a little under a month ago (which is the real cause to the issue you reported.)

DiscordLink 3.0 releases tomorrow after the official Eco 9.5 release and with that this issue should be resolved.

**Comment by mixels** [comment]
Yeah I know. I didn't know Eco 9.5 was coming so soon, so I was excited for a fixed fork of v2. Will look forward to using this again in 9.5! Thanks for the update.

**Comment by MonzUn** [comment]
This issue should be resolved in 3.0.0.
